modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/execution/MethodArgumentResolverDelegate.java

```diff
@@ -126,8 +143,14 @@
    * @param parameterGroupModels {@link List} of {@link ParameterGroupModel} from the corresponding model
    * @param method the {@link Method} to be called
    */
-  public MethodArgumentResolverDelegate(List<ParameterGroupModel> parameterGroupModels, Method method) {
+  public MethodArgumentResolverDelegate(List<ParameterGroupModel> parameterGroupModels, Method method, MuleContext context) {
     this.method = method;
+    this.context = context;
+    try {
+      context.getInjector().inject(this);
```

Comment:
don't, if you need to do this, is much better not to receive the context but use `@Inject` for the three things you need, and then delegate the responsibility of doing the `inject` to the parent of the argument resolver. Remove the context parameter and do the inject in `getMethodArgumentResolver`